nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -104,11 +106,17 @@ def do_compression(
                     mul = opset.reshape(mul, output_shape=original_shape, special_zero=False)
                 last_output = mul.output(0)
             else:
-                compressed_weights, scale, zero_point = _int8_compress(weight, wp.reduction_axes)
-                compressed_const = opset.constant(compressed_weights, dtype=np.uint8, name=weight_name)
-                convert = opset.convert(compressed_const, original_weight_dtype)
-                sub = opset.subtract(convert, zero_point.astype(original_weight_dtype))
+                original_shape = weight.shape
+                compressed_weights, scale, zero_point = _do_integer_quantization(weight, wp.reduction_axes, config)
+                compression_type = np.uint8 if config.num_bits == 8 else ov.Type.u4
+                compressed_weights_node = opset.constant(compressed_weights, dtype=compression_type, name=weight_name)
+                convert_weights_node = opset.convert(compressed_weights_node, original_weight_dtype)
+                zero_point_node = opset.constant(zero_point, dtype=compression_type, name=f"{weight_name}/ZP")
+                convert_zp_node = opset.convert(zero_point_node, original_weight_dtype)
```

Comment:
@andreyanufr @AlexKoff88 `ZP` is whether in UINT8 format or in U4 one. It's always converted to the original precision then.